src/imitation/algorithms/mce_irl.py

```diff
@@ -503,9 +503,11 @@ def train(self, max_iter: int = 1000) -> np.ndarray:
                 predicted_r_np, visitations = self._train_step(torch_obs_mat)
 
                 # these are just for termination conditions & debug logging
-                grad_norm = util.tensor_iter_norm(
-                    p.grad for p in self.reward_net.parameters()
-                ).item()
+                grads = []
+                for p in self.reward_net.parameters():
+                    assert p.grad is not None
+                    grads.append(p)
```

Comment:
Before, the list comprehension added the gradient of the parameter, `p.grad`. Now you're adding the parameter, `p`. This is probably a bug?

Reply:
Oh dear, yes this is a bug, thanks for catching! It's a little sad this was not caught by our tests. :/